### ISSUE TITLE
[Bug] Four implemented loss layers skip `loss_scale` in `calcDerivative`

### ISSUE THREAD

**Issue by kadyrbekovhamit-cyber** [nnstreamer/nntrainer#4329]
## Summary

On current `main` (`a7ea056e79ab8e14447ea305c1b634e233343258`), four
implemented loss derivatives do not apply the `RunLayerContext` loss scale:

- `CrossEntropySigmoidLossLayer`;
- `CrossEntropySoftmaxLossLayer`;
- `KLDLossLayer`;
- `ConstantDerivativeLossLayer`.

`MSELossLayer` does call `LossLayer::applyLossScale`, and the mixed-precision
optimizer path later divides FP32 gradients by the same scale. This makes the
loss-layer behavior inconsistent and can reduce the effective gradient by the
scale factor after unscaling.

## Minimal runtime reproduction

I constructed real `Var_Grad` objects and a real `RunLayerContext` with
`loss_scale=8`, then called the production `calcDerivative` methods.

| Layer | Base derivative | Expected | Observed |
|---|---:|---:|---:|
| sigmoid cross-entropy, `z=0`, label `1` | `-0.5` | `-4.0` | `-0.5` |
| softmax cross-entropy, logits `[0,0]`, label `[1,0]` | `[-0.5,0.5]` | `[-4,4]` | `[-0.5,0.5]` |
| KLD, `P=0.25`, `Q=0.5` | `-0.5` | `-4.0` | `-0.5` |
| constant derivative | `1.0` | `8.0` | `1.0` |

The KLD check was run after locally applying the label-mutation correction
reported in #4328, so that separate defect does not explain this result.

## Minimal fix

Call the existing helper after each derivative is formed:

```cpp
LossLayer::applyLossScale(context, ret_derivative);
```

For KLD the destination tensor is named `deriv`, so the call is:

```cpp
LossLayer::applyLossScale(context, deriv);
```

## Verification

Before the fix, all four focused checks fail with the values above. After the
four helper calls:

```text
loss-scale counterexamples: 4/4 passed
focused loss regression:   71/71 passed
```

The focused regression filter covers both new loss-scale suites, the existing
`LossCross/LayerSemantics` parameterization, and the KLD repeatability test.

I can submit the tests and the four-call patch as a PR if this interpretation
of the `loss_scale` contract matches the maintainers' intent.
